### ISSUE TITLE
Skip the train_step

### ISSUE THREAD

**Issue by Aiah** [allenai/allennlp#5152]
I have an issue that I want to skip a training step and I really cannot filter these instances in DataLoader. I am just wondering if it is possible to skip this batch and carry on with the next one with the allennlp framework?

**Comment by ArjunSubramonian** [comment]
This feature does not currently exist, to the best of my knowledge. Could you explain your need in more detail? Are you running an AllenNLP command and do you want to potentially specify batches to skip as part of overriding the default experiment configuration? 

Additionally, I'm unsure if this would be a good feature. There doesn't seem to be a good use case for skipping particular batches, even if your data were in a deterministic order, unless there are issues with your data, in which case this is more of a dataset problem rather than a data loader problem. However, your elaboration might help me better understand the situation.

**Comment by Aiah** [comment]
Hi, I need to make a multi-label classification task first. Then a generation task is performed on the correctly classified data, so there may be batches that do not have the correctly classified data, and I need to skip this training step. Although it is possible to write the classification results into a file and then read them out so that the data that is not classified correctly can be filtered out in the DataLoader, this method is somewhat troublesome.

**Comment by ArjunSubramonian** [comment]
Oh ok, I see. Are you alternating between the classification and generation tasks? You need to provide more details about your training procedure, and ideally include the high-level code you are running.

**Comment by Aiah** [comment]
Hi, I let the batch without any train data return _None_, then modify train.py and trainer.py. If the batch output is _None_, then skip this batch. I think it works fine now, but I wonder it is correct.
